hw8/task1.py

```diff
@@ -0,0 +1,128 @@
+"""
+We have a file that works as key-value storage, each like is represented as key and value separated by = symbol, example:
+
+name=kek last_name=top song_name=shadilay power=9001
+
+Values can be strings or integer numbers. If a value can be treated both as a number and a string,
+ it is treated as number.
+
+Write a wrapper class for this key value storage that works like this:
+
+storage = KeyValueStorage('path_to_file.txt')
+that has its keys and values accessible as collection items and as attributes.
+Example:
+    storage['name'] # will be string 'kek'
+    storage.song_name # will be 'shadilay'
+    storage.power # will be integer 9001
+
+In case of attribute clash existing built-in attributes take precedence.
+In case when value cannot be assigned to an attribute (for example when there's a line 1=something)
+ ValueError should be raised.
+ File size is expected to be small, you are permitted to read it entirely into memory.
+"""
+from os import path
+from typing import Union
+
+
+class KeyValueStorage:
+    def __init__(self, file_path: str):
+        super().__setattr__("_file_path", file_path)
+
+        if not path.isfile(file_path):
+            raise FileNotFoundError
+        with open(file_path) as f:
+            # use specified dict to prevent rewriting collisions of class attributes from __dict__
+            super().__setattr__("_file_content", {})
+            for line in f:
+                key, value = line.strip().split("=")
+                if value.isdigit():
+                    value = int(value)
+                self._file_content[key] = value
+
+    def __getattr__(self, item: str) -> Union[int, str]:
+        """
+        Get attribute value through . notation
+        """
+        try:
+            res = self._file_content[item]
+        except KeyError:
+            raise AttributeError
+        return res
+
+    def __getitem__(self, item: str) -> Union[int, str]:
+        """
+        Get attribute value via square brackets
+        """
+        if item in self.__dict__:
+            return self.item
+
+        return self._file_content[item]
+
+    def __setattr__(self, name: str, value: Union[int, str]) -> None:
+        """
+        Set attribute value through . notation
+        """
+        if name in self.__dict__:
+            super().__setattr__(name, value)
+            return
+        # a value type check
+        if not isinstance(value, (int, str)):
+            raise ValueError
+
+        file_path = self._file_path
+        file_storage = self._file_content
+
+        self.change_or_delete_line_from_file(file_path, name, value)
```

Comment:
if we want to update the existent value, this line would be called because we will move out the method on 67 line, isn't it?

Reply:
Code on 67 file is expected to manage the case of setting attributes from self._ _dict_ _ which are not from file (there is no alternatives of _ _getattribute_ _ method for setting)